### PR TITLE
Feature/rails 5 updates

### PR DESCRIPTION
**Pull request by TylerRockwell**
### Why?
Addresses #41, final bullet on #47, and #50 

### What Changed?
Use Puma for Rails 5 and Unicorn for Rails 4
Use Tape 2.0 for Rails 5 and 1.8 for Rails 4
Addresses changes in Rubocop rules

### CODE REVIEW

File: lib/gemfile_configurator.rb [hunk 1]
```diff
@@ -13,7 +13,9 @@ def integrated_app_gemfile
 end
 
 def rails_4_gemfile
-  gsub_file 'Gemfile', /5.\d.\d/, '4.2.7.1'
+  gsub_file 'Gemfile', /'rails', '~> \d.*'/, "'rails', '~> #{Rails::VERSION::STRING}'"
+  gsub_file 'Gemfile', /puma/i, 'unicorn'
```
Comment (dacur) [inline lib/gemfile_configurator.rb:17]: what's the `/i` here after puma?

**Comment by dacur** [comment]
thanks for working on this 👍

File: lib/gemfile_configurator.rb (hunk 1 above)
Comment (TylerRockwell) [reply lib/gemfile_configurator.rb:17]: Case insensitive

**Comment by TylerRockwell** [comment]
Happy to help. Thanks for the review, as always @dacur

**Pull request merged**